### PR TITLE
Features: render home page & single game page

### LINKED ISSUE #1: Automate PR template

### LINKED ISSUE #2: On page load, user should see a navbar
- logo
- Navigation Link: (h1 with app name)

### LINKED ISSUE #7: A user should be able to route to a game detail page
By clicking on preview card of game

### LINKED ISSUE #9: Board game detail page contents
- Game title
- Game official rules link
- Game photo
- Some game specs (like # players, time, etc)
- Game details
- Game contested rules (in order of most to least contested)

### LINKED ISSUE #16: Refactor Home data fetching method
https://nextjs.org/docs/api-reference/data-fetching/getInitialProps

instead of getInitialProps:
getStaticProps or getServerSideProps

### LINKED ISSUE #17: Refactor Home to functional component?

### PR DESCRIPTION
**Pull request by hannakim91**
# Team members present: Hanna, Katy, Michael

## What's this PR do? 
* Refactor `Home` component to be a functional component that gets props using `getStaticProps`
* Add function to fetch single game
* Use `getStaticPaths` and `getStaticProps` to set `GameDetails` props dynamically for different games' details pages to render.
* Check functionality of `NavBar` using `Sample` page - this should get replaced with routing back to `Home` on click
- [X] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Styling
- [ ] Testing
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Where should the reviewer start? 
Main changes to:
`pages/index.js`
`pages/api/apiCalls.js`
`pages/GameDetails/[gameid].js`

## How should this be manually tested? 
Run `npm run dev` in terminal and open localhost:3000 in your browser. Should be able to click on each `SingleCard` element and be routed to `GameDetail` page

## Any background context you want to provide? 
We haven't done any testing yet so we should get that done before moving forward with more features. Currently using hardcoded data for `Home` fetch as to not overdo the 60 API calls/minute maximum set by the API we are using.

## What are the relevant tickets? 
* Addresses:
* Closes #1;
* Closes #2;
* Closes #7;
* Closes #9;
* Closes #16;
* Closes #17;

## Screenshots (if appropriate) 
n/a

## Questions:
* Can't seem to make it so that a `GameDetails` page can bring in game data as an object instead of object in an array inside an object by setting `props.game` to evaluate to a variable that evaluates to game.games[0] in `getStaticProps` so it's messier in the render
* Indenting in some of the files are spacing 4 instead of 2 -- how to fix?

### CODE REVIEW

File: components/Cards/Cards.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     const formattedCards = props.games.map((game) => {
-        return <SingleCard
-            image = {game.image}
-            key = {game.id}
-            title = {game.name}
-            />
+    return <SingleCard
+      image={game.image}
+      title={game.name}
+      id={game.id}
```
Comment (MichaelEWalker87) [inline components/Cards/Cards.js:11]: I am not sure of how we can make key and id different or if that is even needed

File: components/NavBar/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <>
+      {/*logo and header that link back to home */}
+      <h1>
+        <Link href="/">
+          <a>House Rules</a>
+        </Link>
+      </h1>
+    </>
+  )
+}
```
Comment (MichaelEWalker87) [inline components/NavBar/NavBar.js:14]: Great work getting the navbar up that was a big lift

**Review by krogowsk531 — APPROVED** [review]
Looks great!  Lots of forward movement here.  Let's begin testing!  Will see if indentation issue is still happening inside of atom as opposed to VS code.

File: pages/GameDetails/[gameid].js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      <img className="game-image"
+        src={game.games[0].image_url}
+        alt={game.games[0].name}
+      />
+      <p className="game-detail-para">{game.games[0].description_preview}</p>
+      <h3 className="game-detail-heading">Number of Players</h3>
+      <p className="game-detail-para">Min: {game.games[0].min_players} - Max: {game.games[0].max_players}</p>
+      <h3 className="game-detail-heading">Estimate Play Time</h3>
+      <p className="game-detail-para">"{game.games[0].max_playtime}</p>
+      <h3 className="game-detail-heading"><a href={game.games[0].rules_url}>Official Game Rules</a></h3>
```
Comment (MichaelEWalker87) [inline pages/GameDetails/[gameid].js:34]: I still think it's odd that it is the index 0 might be something we would check later down the line on this project outside of the scope of this time frame.

File: pages/api/apiCalls.js [hunk 4]
```diff
@@ -4,4 +4,8 @@ export const getGames = async () => {
   const promise =  await fetch(`https://api.boardgameatlas.com/api/search?order_by=popularity&client_id=REACT_APP_KEYCODE`)
   return await promise.json()
 }
-//
\ No newline at end of file
+
+export const getSingleGame = async (gameid) => {
+  const promise = await fetch (`https://api.boardgameatlas.com/api/search?ids=${gameid}&client_id=REACT_APP_KEYCODE`)
```
Comment (MichaelEWalker87) [inline pages/api/apiCalls.js:9]: Good use of the App_KEYCODE

**Review by MichaelEWalker87 — APPROVED** [review]
Great work today I think we have some things to look at like the form but I think this project is on its way

**Pull request merged**